spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/query/SpannerStatementQueryExecutor.java

```diff
@@ -169,32 +171,63 @@ private SpannerStatementQueryExecutor() {
 	 * @param <T> the type of the child persistent entity
 	 * @param writeConverter a converter to convert key values as needed to bind to the query
 	 *     statement.
+	 * @param mappingContext mapping context
 	 * @return the Spanner statement to perform the retrieval.
 	 */
 	public static <T> Statement getChildrenRowsQuery(Key parentKey,
-			SpannerPersistentEntity<T> childPersistentEntity, SpannerCustomConverter writeConverter) {
-		StringBuilder sb = new StringBuilder(
-				"SELECT " + getColumnsStringForSelect(childPersistentEntity) + " FROM "
-						+ childPersistentEntity.tableName() + " WHERE ");
-		StringJoiner sj = new StringJoiner(" and ");
+			SpannerPersistentEntity<T> childPersistentEntity, SpannerCustomConverter writeConverter,
+			SpannerMappingContext mappingContext) {
+		return getChildrenRowsQuery(KeySet.singleKey(parentKey), childPersistentEntity, writeConverter, mappingContext);
+	}
+
+	public static <T> Statement getChildrenRowsQuery(KeySet keySet,
+			SpannerPersistentEntity<T> childPersistentEntity, SpannerCustomConverter writeConverter,
+			SpannerMappingContext mappingContext) {
+		StringJoiner orJoiner = new StringJoiner(" OR ");
 		List<String> tags = new ArrayList<>();
 		List keyParts = new ArrayList();
 		int tagNum = 0;
 		List<SpannerPersistentProperty> childKeyProperties = childPersistentEntity
 				.getFlattenedPrimaryKeyProperties();
-		Iterator parentKeyParts = parentKey.getParts().iterator();
-		while (parentKeyParts.hasNext()) {
-			SpannerPersistentProperty keyProp = childKeyProperties.get(tagNum);
-			String tagName = "tag" + tagNum;
-			sj.add(keyProp.getColumnName() + " = @" + tagName);
-			tags.add(tagName);
-			keyParts.add(parentKeyParts.next());
-			tagNum++;
+
+		for (Key key : keySet.getKeys()) {
```

Comment:
Can you make the for loop and the while loop separate private methods for readability?